metadata-ingestion/scripts/docgen.py

```diff
@@ -676,7 +676,13 @@ def generate(
     sources_dir = f"{out_dir}/sources"
     os.makedirs(sources_dir, exist_ok=True)
 
-    for platform_id, platform_docs in source_documentation.items():
+    i = 0
+    for platform_id, platform_docs in sorted(
+        source_documentation.items(),
+        key=lambda x: (x[1]['name'].casefold(), x[1]['name'])
```

Comment:
This logic is a bit confusing, particularly the accessing of index 1 and index 0  - it's difficult to understand what this structure is supposed to look contain - consider adding a comment?

Reply:
Just took a bit of time to digest. I think I get it :)